java/arcs/core/entity/Handle.kt

```diff
@@ -25,6 +27,20 @@ interface Handle {
     suspend fun close()
 }
 
+data class HandleSpec<T : Entity>(
```

Comment:
What if this were a `sealed class HandleSpec` with subclasses `SignletonHandleSpec` and `CollectionHandleSpec`? Then the constructors could only accept the properly-typed spec, and `init { check...` could be removed.

Reply:
That's a neat idea, but there's going to be other ways of splitting this class up too (e.g. EntityHandleSpec vs. ReferenceHandleSpec) and we won't be able to do that for all of them. I think it's better to treat them all the same.